### PR TITLE
Add riscv64 arch

### PR DESCRIPTION
**Pull request by dongjiang1989**

### CODE REVIEW

File: vars.sh [hunk 1]
```diff
@@ -18,6 +18,11 @@
 IMAGE_NAME="$1"; shift;
 ARCH="$1"; shift;
 SUFFIX="$1"; shift;
+TAG="buster-slim"
+if [ ${ARCH} == "riscv64/" ]
+then
```
Comment (SuperQ) [inline vars.sh:23]:
```suggestion
if [[ ${ARCH} == "riscv64/" ]]; then
```

Comment (dongjiang1989) [reply vars.sh]: @SuperQ Thinks your review. Done.

File: Makefile [hunk 2]
```diff
@@ -60,15 +64,19 @@ manifest:
...
-	@./push.sh "$(REPOSITORY)/$(NAME)-linux-armv7" "arm32v7" "$(SUFFIX)" $(VERSIONS)
-	@./push.sh "$(REPOSITORY)/$(NAME)-linux-arm64" "arm64v8" "$(SUFFIX)" $(VERSIONS)
+	@./push.sh "$(REPOSITORY)/$(NAME)-linux-armv7" "arm32v7/" "$(SUFFIX)" $(VERSIONS)
+	@./push.sh "$(REPOSITORY)/$(NAME)-linux-arm64" "arm64v8/" "$(SUFFIX)" $(VERSIONS)
+	# glibc doens't support riscv64
+	@./push.sh "$(REPOSITORY)/$(NAME)-linux-riscv64" "riscv64/" "$(SUFFIX)" uclibc
 	# uclibc doens't support ppc64le, s390x
-	@./push.sh "$(REPOSITORY)/$(NAME)-linux-ppc64le" "ppc64le" "$(SUFFIX)" glibc
-	@./push.sh "$(REPOSITORY)/$(NAME)-linux-s390x" "s390x" "$(SUFFIX)" glibc
+	@./push.sh "$(REPOSITORY)/$(NAME)-linux-riscv64" "riscv64/" "$(SUFFIX)" glibc
```
[the remark this answers is not in the document]
Comment (dongjiang1989) [reply Makefile:80]: Thinks your review. Done.

File: vars.sh [hunk 3]
```diff
@@ -18,8 +18,12 @@
 IMAGE_NAME="$1"; shift;
 ARCH="$1"; shift;
 SUFFIX="$1"; shift;
+TAG="buster-slim"
+if [[ "${ARCH}" == "riscv64/" ]]; then
+    TAG="sid-slim"
+fi
 
-if [[ -n "${ARCH}" ]]; then
+if [[ -n "${ARCH}" && "${ARCH}" != "" ]]; then
```
[the remark this answers is not in the document]
Comment (dongjiang1989) [reply vars.sh]: Done. Double checker. this's bug

**Comment by dongjiang1989** [comment]
@SuperQ  PTAL!

File: Makefile [hunk 4]
```diff
@@ -60,15 +64,18 @@ manifest:
...
 		"$(REPOSITORY)/$(NAME)-linux-ppc64le:latest" \
 		"$(REPOSITORY)/$(NAME)-linux-s390x:latest"
 	DOCKER_CLI_EXPERIMENTAL=enabled docker manifest push "$(REPOSITORY)/$(NAME):latest"
 
 .PHONY: push
 push:
 	@./push.sh "$(REPOSITORY)/$(NAME)-linux-amd64" "" "$(SUFFIX)" $(VERSIONS)
-	@./push.sh "$(REPOSITORY)/$(NAME)-linux-armv7" "arm32v7" "$(SUFFIX)" $(VERSIONS)
-	@./push.sh "$(REPOSITORY)/$(NAME)-linux-arm64" "arm64v8" "$(SUFFIX)" $(VERSIONS)
+	@./push.sh "$(REPOSITORY)/$(NAME)-linux-armv7" "arm32v7/" "$(SUFFIX)" $(VERSIONS)
```
Comment (SuperQ) [inline Makefile:75]: This still needs to be fixed.

Comment (dongjiang1989) [reply Makefile]: Sorry! Lost, Now done

File: uclibc/Dockerfile [hunk 5]
```diff
@@ -1,5 +1,6 @@
 ARG ARCH=""
-FROM ${ARCH}debian:bullseye-slim
+ARG TAG="buster-slim"
```
[the remark this answers is not in the document]
Comment (dongjiang1989) [reply uclibc/Dockerfile:2]: Too bad there is no official package. https://hub.docker.com/r/riscv64/debian/tags

[the remark this answers is not in the document]
Comment (dongjiang1989) [reply uclibc/Dockerfile:2]: > I just fixed this to use bullseye.

Too bad there is no official package. https://hub.docker.com/r/riscv64/debian/tags

[the remark this answers is not in the document]
Comment (SuperQ) [reply uclibc/Dockerfile:2]: Yea, maybe once bookworm is out.

[the remark this answers is not in the document]
Comment (dongjiang1989) [reply uclibc/Dockerfile]: @SuperQ  Setting default tag `bullseye-slim`. Please review

**Comment by discordianfish** [comment]
@dongjiang1989 This needs rebase

**Comment by discordianfish** [comment]
@dongjiang1989 sorry man but looks like this needs another rebase.

**Pull request closed without merging**